### DISCUSSION TITLE
Input voltage boost

### DISCUSSION THREAD

**Discussion by MeisyarohAzzahra** [fhdm-dev/scoppy#104] [Q&A]
Can RPI Pico be combined with ADS1115 module to be able to measure 5V input voltage? By using certain code on Microphyton

**Comment by JRGould** [comment]
All you really need here is a voltage divider.

Just connect the pico's adc input pin to two resistors, `R1` and `R2` with the  other end of `R1` connected to your input probe and the other end of R2 connected to ground. 

```
       ┌──────────┐        ┌──────────┐
     ┌─◍ R1 (2mΩ) ◍───┬────◍ R2 (3mΩ) ◍──┐
     │ └──────────┘   │    └──────────┘  │
     │                │                  │
     ◍                ◍ (~3V)            │
     │                │                  │
    ┌▼┐             ┌─▼┐                 │
┌───┤G├─────────────┼A0├──────┐          │
│   └─┘             └──┘      │          │
│        rpi pico             │          │    ╔═════════════╗
│                             │          └────◇    Probe    ║────◎ (~5v)
└─────────────────────────────┘               ╚═════════════╝

```

You can use [this calculator](https://ohmslawcalculator.com/voltage-divider-calculator) and set `Vout` to 3.3v (the pico's max adc input), and use `Vs` as the maximum voltage you might probe. For 5volts, I'd probably use `R1=2mΩ` and `R2=3mΩ` which would max you out at 5.5V.

There's also a setting in Scoppy that lets you choose the voltage range for the input, so you'd just use min=0 and max=`Vs` where `Vs` is the value the voltage divider calculator gives you for your chosen resistors when `Vout` is 3.3v

**Comment by MeisyarohAzzahra** [comment]
means if i wanna increase the voltage source then i hafta always change the value of the resistor?
